File: patient.py

# pip install lifelines
# import lifelines

import pandas as pd
# Loading the the survival un-employment data
patient = pd.read_csv("C:\\OneDrive\\Desktop\\360digiTMG assignment\\Survival analytics\\Patient.csv")
patient.head()
patient.describe()

from sklearn.preprocessing import LabelEncoder
# creating instance of labelencoder
labelencoder = LabelEncoder()

patient.columns
patient['PatientID']= labelencoder.fit_transform(patient['PatientID'])

patient["PatientID"].describe()

# PatientID is referring to time 
T = patient.PatientID

# Importing the KaplanMeierFitter model to fit the survival analysis
from lifelines import KaplanMeierFitter

# Initiating the KaplanMeierFitter model
kmf = KaplanMeierFitter()

# Fitting KaplanMeierFitter model on Time and Eventtypes for death 
kmf.fit(T, event_observed=patient.Eventtype)

# Time-line estimations plot 
kmf.plot()

# Over Multiple groups 
# For each group, here group is Followup
patient.Followup.value_counts()

# Applying KaplanMeierFitter model on Time and Eventtypes for the group "1"
kmf.fit(T[patient.Followup==1], patient.Eventtype[patient.Followup==1], label='1')
ax = kmf.plot()

# Applying KaplanMeierFitter model on Time and Eventtypes for the group "0"
kmf.fit(T[patient.Followup==0], patient.Eventtype[patient.Followup==0], label='0')
kmf.plot(ax=ax)

File: patient.R
# Survival Analysis in R

install.packages('survminer')
install.packages("survival")

library(survminer)
library(survival)

??survminer
??survival

patient <- read.csv("C:\\Users\\Desktop\\360digiTMG assignment\\Survival analytics\\Patient.csv")
patient$PatientID <-factor(patient$PatientID)

patient$PatientID <-as.numeric(patient$PatientID)
attach(patient)
str(patient)

# Define variables 
time <- PatientID
event <- Eventtype
group <- Followup
# Descriptive statistics
summary(time)
table(event)
table(group)

# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time, event) ~ 1)

summary(kmsurvival)

plot(kmsurvival, xlab="Time", ylab="Survival Probability")

ggsurvplot(kmsurvival, data=patient, risk.table = TRUE)


# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ group)
summary(kmsurvival1)

plot(kmsurvival1, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=patient, risk.table = TRUE)

###############
